src/Tickets/Emails/Email_Template.php

```diff
@@ -0,0 +1,186 @@
+	
+	/**
+	 * Gets the template instance used to setup the rendering html.
+	 *
+	 * @since TBD
+	 *
+	 * @return Tribe__Template The template object.
+	 */
+	public function get_template(): Tribe__Template {
+		if ( empty( $this->template ) ) {
+			$this->template = new Tribe__Template();
+			$this->template->set_template_origin( Tribe__Tickets__Main::instance() );
+			// @todo Move template folder into `src/views/v2` before TE release.
+			$this->template->set_template_folder( 'src/admin-views/settings/emails' );
+			$this->template->set_template_context_extract( true );
+		}
+
+		return $this->template;
+	}
+
+	/**
+	 * Returns the email template HTML.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The HTML of the template.
+	 */
+	public function get_html() {
+		$template = $this->get_template();
+		return $template->template( 'email-template', $this->get_context(), false );
+	}
+
+	/**
+	 * Prints the email template HTML.
+	 *
+	 * @since TBD
+	 *
+	 * @return void.
+	 */
+	public function render() {
+		echo $this->get_html();
+	}
+
+	/**
+	 * Sets whether or not this will be a template preview.
+	 *
+	 * @since TBD
+	 *
+	 * @param boolean $is_preview
+	 * 
+	 * @return void
+	 */
+	public function set_preview( $is_preview = false ) {
+		$this->preview = $is_preview;
+	}
+
+	/**
+	 * Is this a template preview?
+	 *
+	 * @since TBD
+	 *
+	 * @return boolean Whether or not this is a template preview.
+	 */
+	public function is_preview() {
+		return $this->preview;
+	}
+
+	/**
+	 * Sets the data for the template context.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $data
+	 * 
+	 * @return void
+	 */
+	public function set_data( array $data ) {
+		$this->context_data = $data;
+	}
+
+	/**
+	 * Returns the template context array and creates sample data if preview.
+	 *
+	 * @since TBD
+	 *
+	 * @return array Template context array.
+	 */
+	public function get_context() {
+		$context = [
+			'preview'                => $this->preview,
+			'header_image_url'       => tribe_get_option( Settings::$option_header_image_url, '' ),
+			'header_image_alignment' => tribe_get_option( Settings::$option_header_image_alignment, 'left' ),
+			'header_bg_color'        => tribe_get_option( Settings::$option_header_bg_color, '#ffffff' ),
+			'ticket_bg_color'        => tribe_get_option( Settings::$option_ticket_bg_color, '#ffffff' ),
+			'footer_content'         => tribe_get_option( Settings::$option_footer_content, '' ),
+			'footer_credit'          => tribe_get_option( Settings::$option_footer_credit, true ),
+		];
+		$context['header_text_color'] = Tribe__Utils__Color::get_contrast_color( $context['header_bg_color'] );
+		$context['ticket_text_color'] = Tribe__Utils__Color::get_contrast_color( $context['ticket_bg_color'] );
+
+		if ( $this->preview ) {
+			$this->context_data = $this->get_preview_context_array();
+		}
+		
+		$this->context_data = apply_filters( 'tec_tickets_emails_email_template_context', array_merge( $context, $this->context_data ) );
```

Comment:
We're calling the filter two times, we need to avoid that:
```suggestion
		$this->context_data = array_merge( $context, $this->context_data );
```